### PR TITLE
10_simplification exercise

### PR DESCRIPTION
**Pull request by riraum**
Exercise 10_simplification

### CODE REVIEW

File: 10_simplification/main.go [hunk 1]
```diff
@@ -145,31 +147,122 @@ can be taken.
...
+		if k%2 == 0 {
+			fmt.Println(k)
+		}
+	}
 	// TODO: write a loop that prints the first 10 odd numbers
+	for l := 0; l < 10; l++ {
+		if l%2 != 0 {
+			fmt.Println(l)
+		}
+	}
```
Comment (nobe4) [inline 10_simplification/main.go]: This prints the first odd numbers under 10, the task is to print the first 10 odd numbers.

I.e., you print 1, 3, 5, 7, 9
You need to print 1, 3, 7, 9, 11, 13, 15, 17, 19, 21

File: 10_simplification/main.go [hunk 2]
```diff
@@ -145,31 +147,122 @@ can be taken.
...
+		sumFirstTen += m
+	}
+	// debug
+	fmt.Println(sumFirstTen)
 	// TODO: write a loop that calculates the sum of the first 10 odd numbers
+	var sumFirstTenOdd int
+	for n := 0; n < 10; n++ {
+		if n%2 != 0 {
+			sumFirstTenOdd += n
+		}
```
Comment (nobe4) [inline 10_simplification/main.go]: Same comment as https://github.com/riraum/forward/pull/14/files#r1814380788

File: 10_simplification/main.go [hunk 3]
```diff
@@ -145,31 +147,122 @@ can be taken.
...
+// parameters.
+// If the second parameter is less than the first, return 0.
+// e.g. sum(1, 3) => 1 + 2 + 3 = 6
+//
+//	sum(2, 5) => 2 + 3 + 4 + 5 = 14
+func returnBetween(a, b float64) float64 {
+	var sum float64
+	if b < a {
+		return 0
+	}
```
Comment (nobe4) [inline 10_simplification/main.go]: You only need `sum` if you use it, so you can define it later

```suggestion
	if b < a {
		return 0
	}
	
	var sum float64
```

File: 10_simplification/main.go (hunk 1 above)
Comment (riraum) [reply 10_simplification/main.go]: Done!

File: 10_simplification/main.go (hunk 2 above)
Comment (riraum) [reply 10_simplification/main.go]: Done!

File: 10_simplification/main.go (hunk 3 above)
Comment (riraum) [reply 10_simplification/main.go]: Done!

File: 10_simplification/main.go [hunk 4]
```diff
@@ -145,31 +147,129 @@ can be taken.
...
+		}
+	}
 	// TODO: write a loop that prints the first 10 odd numbers
+	x := 0
+	for l := 0; x <= 10; l++ {
+		if l%2 == 0 {
+			continue
+		}
+		fmt.Println("Print odd numbers", l)
+		x++
```
Comment (nobe4) [inline 10_simplification/main.go]: This is too complex, you don't need `x` and `l`, you can do with just one variable.

File: 10_simplification/main.go [hunk 5]
```diff
@@ -145,31 +147,129 @@ can be taken.
...
 	// TODO: write a loop that calculates the sum of the first 10 odd numbers
+	var sumFirstTenOdd int
+	y := 0
+	for n := 0; y < 10; n++ {
+		if n%2 == 0 {
+			continue
+		}
+		sumFirstTenOdd += n
+		y++
+	}
```
Comment (nobe4) [inline 10_simplification/main.go]: Same comment here.

File: 10_simplification/main.go (hunk 4 above)
Comment (riraum) [reply 10_simplification/main.go]: Took me quite a bit of playing around. Did try various counter placements, no dice.
Had to search the web and after no result seemed to help, read the Brave Search AI answer, that helped.
I guess I do have to create the logic around that after an odd number is followed by an odd number and therefore both increase my break limit and counter to double the values.
Screenshot 2024-10-24 at 20 05 48
I try to stay away from AI for now, could disable the auto results if you think that's better.

File: 10_simplification/main.go (hunk 5 above)
Comment (riraum) [reply 10_simplification/main.go]: Same as above, should be done.

File: 10_simplification/main.go [hunk 6]
```diff
@@ -171,14 +171,13 @@ func main() {
 		}
 	}
 	// TODO: write a loop that prints the first 10 odd numbers
-	x := 0
-	for l := 0; x <= 10; l++ {
-		if l%2 == 0 {
-			continue
+	for l := 0; l <= 20; l++ {
```
Comment (nobe4) [inline 10_simplification/main.go:174]: That's not what I had in mind.

Look again at how you solved it in 05_loop.

File: 10_simplification/main.go (hunk 4 above)
Comment (nobe4) [reply 10_simplification/main.go]: That's not what I had in mind.

Look again at how you solved it in 05_loop.

Comment (riraum) [reply 10_simplification/main.go]: My implementation differed a bit from the Brave Search result.
I think I had looked at 05_loops and did check the `Write a loop that prints the first 100 odd numbers.` again, but there I use 2 variables.
Added that logic too, but with 2 variables.

I did try to remove the continue condition with from this shorter loop syntax, but seems it has to stay there, seem to only be able to add a break statement after.

Comment (nobe4) [reply 10_simplification/main.go]: Nice, it's good that you did a couple of additional examples!

**Review by nobe4 — APPROVED** [review]

**Pull request merged**